### DISCUSSION TITLE
🤔 🔍 Where can I find the ETHOS acknowledgements/licenses of FOSS subcomponents etc.?

### DISCUSSION THREAD

**Discussion by packetpilot** [FrSkyRC/ETHOS-Feedback-Community#3280] [Q&A]
I've checked the ethos subdomain, and the Ethos Suite software, and my radio (when it had ethos installed) and was surprised to not see any acknowledgements/licenses of subcomponents etc. Even the ethos manual linked by the suite didn't have any mention of this sort of info.

Is this because it's somewhere I've not looked, or because it is 100% original software that in no way stands on the shoulders of other projects? ...or something else?

**Comment by bsongis-frsky** [comment]
Ethos relies on libopenui and ST libs. That's all.
- for libopenui, I own it, I suppose I can decide not to bother about it
- for ST libs I have to check, if really needed it will be done, at least in the EN manual. It could be done in Ethos Suite too

Ethos Suite relies on Electron and Angular/Material, I see nothing is mandatory in the Electron license, I don't think it's different for Angular / Material components. When working with such a platform you reuse a ton of components that you don't even know. I should check how other closed source projects deal with that IP issue.

Feel free to say I am wrong on anything above!

**Reply by packetpilot** [reply]
> When working with such a platform you reuse a ton of components that you don't even know.

I get what you're saying, but finding transitive dependencies is not difficult. Getting your SCM to populate all the dependencies' -- and transitive dependencies' -- licenses into a single file that is rendered _in the product_ (in other words, on the radio) can certainly be difficult. Yes, transitive dependencies count, too.

Admittedly there are big, widely-adopted projects written/maintained by authors who simply don't care to be diligent about this. Rarely are those projects _commercial_, which ethos certainly is. I can't claim to be an expert on the sorts of repercussions that are possible, but based on how seriously a couple of the "internet scale" companies I've worked for were/presumably still are these things, the consequences can be "bad" for companies who've decided not to care.

I suppose nobody's paying for ethos, but based on the ethos subdomain, the OS itself is clearly being used to market and sell the radios, and of course the OS is not available in any form (opaque binary or otherwise) for radios that are not FrSky radios. People are certainly choosing FrSky radios for the OS. Of course, this is great -- your team's doing good work! 🎉 ...but it seems there's some growing up to do w/r/t one of the most truly boring parts of writing commercial software.

> for libopenui, I own it, I suppose I can decide not to bother about it

That's just not how any of this works. Rather go on a diatribe about the Kantian maxim, I'd just point out that the list of software acknowledged by another piece of software is not a list containing only those who've bothered to whine about not being acknowledged.

I'd be willing to bet my last three months income that there are GPLv_N_ violations taking place, so I'll end with a [page](https://www.gnu.org/licenses/quick-guide-gplv3.en.html) on gnu.org:

> Under GPLv2, if you violated the license in any way, your rights were automatically and permanently lost. The only way to get them back was to petition the copyright holder. While a strong defense against violations is valuable, this policy could cause a lot of headache when someone accidentally ran afoul of the rules. Asking all the copyright holders for a formal restoration of the license could be burdensome and costly: a typical GNU/Linux distribution draws upon the work of thousands.
> 
> GPLv3 offers a reprieve for good behavior: if you violate the license, you'll get your rights back once you stop the violation, unless a copyright holder contacts you within 60 days. After you receive such a notice, you can have your rights fully restored if you're a first-time violator and correct the violation within 30 days. Otherwise, you can work out the issue on a case-by-case basis with the copyright holders who contacted you, and your rights will be restored afterward.
> 
> Compliance with the GPL has always been the top priority of the FSF Compliance Lab and other groups enforcing the license worldwide. These changes ensure that compliance remains the top priority for enforcers, and gives violators incentive to comply.

**Reply by bsongis-frsky** [reply]
> licenses into a single file that is rendered in the product (in other words, on the radio)

Not on the radio, it's Ethos Suite, which runs on a PC / Mac

**Comment by bsongis-frsky** [comment]
Issue opened here to be sure we are on the right side:
https://github.com/FrSkyRC/ETHOS-Feedback-Community/issues/3290

Then you prepare your last three months income, I will send you my IBAN when you are ready ;)

**Comment by bsongis-frsky** [comment]
Issue fixed in 1.5.0 which is currently in RC.
